router/router.go

```diff
@@ -81,3 +81,14 @@ func newFromConfigBytes(fileBytes []byte) (RuleRouter, error) {
 	}
 	return router, err
 }
+
+// NewFromRoutes constructs a RuleRouter using the provided map of route names
+// to Rules.
+func NewFromRoutes(routes map[string]Rule) RuleRouter {
```

Comment:
👍 

I think you could refactor the above methods to use this. I would combine `NewFromConfig` and `newFromConfigBytes` into one method... It should open a file, and yaml.Unmarshal the bytes into routes... then pass the routes into `NewFromRoutes`.